docs/csharp/tutorials/generate-consume-asynchronous-stream.md

```diff
@@ -0,0 +1,120 @@
+---
+title: Generate and consume async streams
+description: This advanced tutorial illustrates scenarios where generating and consuming async streams provides a more natural way to work with sequences of data that may be generated asynchronously.
+ms.date: 02/10/2019
+ms.custom: mvc
+---
+# Tutorial: Generate and consume async streams using C# 8.0 and .NET Core 3.0
+
+C# 8.0 introduces **async streams**, which model a streaming source of data when the elements in the data stream may be retrieved or generated asynchronously. Async streams rely on new interfaces introduced in .NET Standard 2.1 and implemented in .NET Core 3.0 to provide a natural programming model for asynchronous streaming data sources.
+
+In this tutorial, you'll learn how to:
+
+> [!div class="checklist"]
+> * Create a data source that generates a sequence of data elements asynchronously.
+> * Consume that data source asynchronously.
+> * Recognize when the new interface and data source are preferred to earlier synchronous data sequences.
+
+## Prerequisites
+
+You’ll need to set up your machine to run .NET Core, including the C# 8.0 beta compiler. The C# 8 beta compiler is available with [Visual Studio 2019 preview 1](https://visualstudio.microsoft.com/vs/preview/), or [.NET Core 3.0 preview 1](https://dotnet.microsoft.com/download/dotnet-core/3.0). Async streams are first available in .NET C.0 preview 2.
+
+This tutorial assumes you're familiar with C# and .NET, including either Visual Studio or the .NET Core CLI.
+
+## Run the starter application
+
+You can get the code for the starter application use in this tutorial from our [samples](https://github.com/dotnet/samples) repository in the [csharp/tutorials/AsyncStreams](https://github.com/dotnet/samples/tree/master/csharp/tutorials/AsyncStreams/start) folder.
+
+The starter application is a console application that uses the[GitHub GraphQL](https://developer.github.com/v4/) interface to retrieve recent issues written in the [.NET docs](https://github.com/dotnet/docs) repository. Start by looking at the code for the starter app `Main` method as shown in the following code:
+
+[!code-csharp[StarterAppMain](../../../samples/csharp/tutorials/AsyncStreams/start/IssuePRreport/IssuePRreport/Program.cs#StarterAppMain)]
+
+You'll need to create a [GitHub access token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/#creating-a-token) so that you can access the GitHub GraphQL endpoint. You can either set a `GitHubKey` environment variable to your personal access token, or you can replace the last argument in the call to `GenEnvVariable` with your personal access token.
```

Comment:
Should this be listed in the prereqs?

I imagine you're using the first technique here for safety reasons. Might be worth mentioning that's the one being used instead of GenEnvVariable.